### PR TITLE
Contentful parser

### PR DESCRIPTION
**Pull request by inigomarquinez**
## Description

This PR adds a mapper to the Contentful content fetch to provide an agnostic output to the user.

It also modifies the existing tests to cover more complex use cases (new content models added to the contentful test dataset).

## Related Issue

-

## Motivation and Context

Make the contentful provider functional.

## How Has This Been Tested?

Existing unit and e2e tests modified.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: src/cms/contentful.ts [hunk 1]
```diff
@@ -1,12 +1,21 @@
...
+  EntryQueries,
 } from 'contentful';
 
+import { HarmonizedOutput } from '../@types';
 import { AbstractAgnosticCMSHarmonizerClient } from '../index.abstract';
 
+enum ContentfulResourceType {
+  ENTRY = 'Entry',
+  ASSET = 'Asset',
+}
```
Comment (matyasjay) [inline src/cms/contentful.ts:17]: Use `const` `as const` please! `enum`s are known to have bad performance and confusing output while providing no value over `const`.

```suggestion
const ContentfulResourceType  = {
  ENTRY: 'Entry',
  ASSET: 'Asset',
} as const
```

Also, I think it is a good idea to decouple consts, magics and any other hard coded data from the implementation and put to a centralised place instead.

**Review by matyasjay — CHANGES_REQUESTED** [review]

File: src/cms/contentful.ts (hunk 1 above)
Comment (inigomarquinez) [reply src/cms/contentful.ts:17]: Done!

About moving to a centralized place, as this is very specific to Contentful and not used out of that class (not needed to be exported to the user), I don't know if it needs to be moved. What place do you suggest @matyasjay ?

Comment (matyasjay) [reply src/cms/contentful.ts:17]: Nah, I wasn't really aware of the scope. Fine for me to keep it here, thanks!

**Review by matyasjay — APPROVED** [review]

File: src/cms/contentful.ts [hunk 2]
```diff
@@ -22,10 +31,48 @@ export class Contentful extends AbstractAgnosticCMSHarmonizerClient {
...
 
-  public async getEntry(entryId: string) {
+  public async getEntry(
+    entryId: string,
+    locale?: string,
+  ): Promise<HarmonizedOutput> {
     return await this.getEntryHarmonized(
-      () => this.getClientInstance().getEntry(entryId),
-      ({ fields }: { fields: any }) => ({ data: fields }),
+      this.getEntryHandler.bind(this, entryId, { locale, include: 10 }),
```
[the remark this answers is not in the document]
Comment (inigomarquinez) [reply src/cms/contentful.ts]: Added as a configurable param

**Review by matyasjay — APPROVED** [review]

**Review by Bounteous17 — APPROVED** [review]

**Pull request merged**